include/pisa/wand_data.hpp

```diff
@@ -22,7 +23,7 @@ namespace pisa {
 
         template<typename LengthsIterator>
         wand_data(LengthsIterator len_it, uint64_t num_docs,
-                      binary_freq_collection const &coll, partition_type type = partition_type::fixed_blocks, const float lambda = 0.0f) {
+                      binary_freq_collection const &coll, partition_type type = partition_type::fixed_blocks, boost::variant<float, uint64_t> block_size = uint64_t(0)) {
```

Comment:
In general, I'd like to use `std::variant` instead. But seeing as there are some problems with its support by AppleClang (lol, I know), we might wanna use boost for now. @amallia Are we ready to use `std::variant` properly or not yet?

Reply:
Not yet, I asked @JMMackenzie to use `boost` because of the incompatibility.
We can migrate all the `boost::variant` occurrences all together at some point.